Stratis.Bitcoin/Features/IndexStore/Index.cs

```diff
@@ -0,0 +1,421 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Collections.Generic;
+using DBreeze.Utils;
+using NBitcoin;
+using Microsoft.CodeAnalysis.CSharp.Scripting;
+using Microsoft.CodeAnalysis.Scripting;
+using Newtonsoft.Json;
+
+namespace Stratis.Bitcoin.Features.IndexStore
+{
+    public class IndexExpression
+    {
+        public static string[] DefaultDependancies = new string[] {
+                    "System", "System.Linq", "System.Linq.Expressions", "System.Collections.Generic", "NBitcoin" };
+
+        [JsonProperty(PropertyName = "Builder")]
+        public string Builder { get; private set; }
+        [JsonProperty(PropertyName = "Many")]
+        public bool Many { get; private set; }
+        [JsonProperty(PropertyName = "Uses")]
+        public string[] Dependencies { get; private set; }
+        [JsonIgnore()]
+        public Expression<Func<Transaction, Block, Network, IEnumerable<object[]>>> expression { get; private set; }
+        [JsonIgnore()]
+        public Func<Transaction, Block, Network, IEnumerable<object[]>> compiled = null;
+
+        public IndexExpression(bool multiValue, string builder, string[] dependencies = null)
+        {
+            if (dependencies == null)
+                dependencies = DefaultDependancies;
+            this.Builder = builder;
+            this.Dependencies = dependencies;
+            this.Many = multiValue;
+        }
+
+        public void Compile()
+        {
```

Comment:
From your later comment in the slack it appears you have figured this out. As you have noted it is used for building C# expressions. I.e. given a string representing a LINQ expression it then compiles that LINQ expression. The use case is to take the "builder" string and produce a usable LINQ expression that can act on the (transaction, block, network) triplet to produce key/value rows to add to an index.